File: 1-TwoSum/solution.py
from typing import List
'''
Given an array of integers nums and an integer target, return indices
of the two numbers such that they add up to target.

You may assume that each input would have exactly one solution, and
you may not use the same element twice.

You can return the answer in any order.

[Runtime] 64ms - Beats 82.49%
Aug 29, 2023
'''


def twoSum(self, nums: List[int], target: int) -> List[int]:
    val = set(nums)
    for count, i in enumerate(nums):
        diff = target - i
        if diff in val:
            secondIndex = nums.index(diff)
            if count != secondIndex:
                return [count, secondIndex]

File: 1-TwoSum/solution.go
package main

import (
    "fmt"
)

func main() {
    nums := []int{2,7,3,5,8}
    k := 9
    answer := findTwoSum(nums, k)
    fmt.Println(answer)
}

func findTwoSum(nums []int, target int) []int {
    visited := make(map[int] int)
    for count, val := range nums {
        if idx, found := visited[target - val]; found {
            return []int{count, idx}
        }
        visited[val] = count
    }
    return nil
}
